File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    participant.sort()
    completion.sort()

    total = len(participant)
    
    for i in range(total - 1):
        if (participant[i] != completion[i]):
            return participant[i]
    
    # 리스트의 마지막 원소에 해당하는 선수가 완주하지 못한 경우 return
    return participant[total - 1]

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.HashMap;

class Solution {
    public String solution(String[] participant, String[] completion) {
        HashMap<String, Integer> players = new HashMap<>();
        
        for (String p: participant) {
            if (players.containsKey(p)) {
                players.put(p, players.get(p) + 1);
            } else {
                players.put(p, 1);
            }
        }
        
        for (String c: completion) {
            players.put(c, players.get(c) - 1);
            if (players.get(c) == 0) {
                players.remove(c);
            }
        }
        
        System.out.println(players.keySet());
        
        
        String answer = "";
        for (String key: players.keySet()) {
            answer = key;
        }
        
        return answer;
    }
}